scanner/datastore/postgres/distributions.go

```diff
@@ -2,15 +2,22 @@ package postgres
 
 import (
 	"context"
+	"fmt"
+	"regexp"
 
 	"github.com/quay/claircore"
+	"github.com/quay/zlog"
 )
 
+var rhelCPE = regexp.MustCompile(`cpe:2\.3:o:redhat:enterprise_linux:(\d+):\*:\*:\*:\*:\*:\*:\*`)
+
 // Distributions retrieves the currently known distributions from the database.
 //
 // A distribution is considered known if there exists at least one row in the vuln table which references it.
 func (m *matcherStore) Distributions(ctx context.Context) ([]claircore.Distribution, error) {
-	const selectDists = `SELECT DISTINCT dist_id, dist_version_id, dist_version FROM vuln`
+	ctx = zlog.ContextWithValues(ctx, "component", "datastore/postgres/distributions/Distributions")
+
+	const selectDists = `SELECT DISTINCT dist_id, dist_version_id, dist_version, repo_name FROM vuln WHERE repo_name = '' OR repo_name LIKE 'cpe:2.3:o:redhat:enterprise_linux:%:*:*:*:*:*:*:*'`
```

Comment:
What is the purpose for limiting these results by repo name, specifically the `enterprise_linux` one? Seems like there are many different redhat `rhel` repos that could provide version details.

Such as:
```
cpe:/a:redhat:rhel_eus:...
cpe:/o:redhat:rhel_eus:...
cpe:/a:redhat:rhel_e4s:...
cpe:/o:redhat:rhel_e4s:...
cpe:/a:redhat:rhel_tus:...
cpe:/o:redhat:rhel_tus:...
cpe:/a:redhat:enterprise_linux:...
cpe:/o:redhat:enterprise_linux:...

etc.
```

Reply:
so the new VEX data is no longer RHEL version specific like OVAL was (OVAL put each file into a RHEL-version-specific directory while VEX is year-based). So, the old query stopped working for RHEL because the vulns are no longer attributed with a RHEL version.

`cpe:2.3:o:redhat:enterprise_linux:%:*:*:*:*:*:*:*` represents the format of the CPE which will give us the RHEL version.

Breaking the [CPE](https://nvlpubs.nist.gov/nistpubs/Legacy/IR/nistir7695.pdf) down it says we are looking for:

Part: Operating System
Vendor: Red Hat
Product: Enterprise Linux
Version: <anything>

So, this format specifies the GA OS (opposed to TUS, EUS, etc), which is what we are interested in. I checked this a while ago when I originally did it, and I think it got me RHEL 2 - 9, but I don't recall off the top of my head

Reply:
yeah, exactly. It's just for notes